README.md

```diff
@@ -534,22 +550,26 @@ spec:
   versions:
   - version: "*"
     containers:
-    - .spec.template.spec.containers
-    - .spec.template.spec.initContainers
-    volumes: .spec.template.spec.volumes
+    - path: .spec.template.spec.containers[*]
+      name: .name
+      env: .env
+      volumeMounts: .volumeMounts
+    - path: .spec.template.spec.initContainers[*]
+      name: .name
+      env: .env
+      volumeMounts: .volumeMounts
+    volumes: .spec/template.spec.volumes
 ```
 
-## Reconciler Implementation
-
-A reconciler implementation **MUST** support mapping to PodSpec-able resources without defining a Workload Resource Mapping for those types.  If no Workload Resource Mapping exists for the `ServiceBinding` workload resource type and the workload resource is not PodSpec-able, the reconciliation **MUST** fail.
+Note: this example is equivalent to not specifying a mapping, or specifying an empty mapping.
 
-If a `ClusterWorkloadResourceMapping` defines `containers`, the reconciler **MUST** first resolve a set of candidate locations in the workload resource addressed by the `ServiceBinding` using the `Container` type (`.envs`, `.volumeMounts`) for all available containers and then filter that collection by the `ServiceBinding` `.spec.workload.containers` filter before applying the appropriate modification.
+## Reconciler Implementation
 
-If a `ClusterWorkloadResourceMapping` defines `.envs` and `.volumeMounts`, the reconciler **MUST** first resolve a set of candidate locations in the workload resource addressed by the `ServiceBinding` for all available containers and then filter that collection by the `ServiceBinding` `.spec.workload.containers` filter before applying the appropriate modification.
+A reconciler implementation **MUST** support mapping to PodSpec-able resources without defining a `ClusterWorkloadResourceMapping` for those types.
 
-If a `ServiceBinding` specifies `.spec.workload.containers` value, that value **MUST** be used to filter all entries in the `.containers` list.  If a `ServiceBinding` specifies a `.spec.workload.containers` value and `ClusterWorkloadResourceMapping` for the mapped type defines `.envs` and `.volumeMounts`, the reconciler **MUST** fail to reconcile.
+If a `ServiceBinding` specifies `.spec.workload.containers` and a `MappingContainer` specifies a `name` expression, the resolved name **MUST** limit which containers in the workload are bound.
```

Comment:
What happens when `name` expression is not specified, but `.spec.workload.containers` is there?

Reply:
In that case there is no requirement to restrict binding to the containers specified by `.spec.workload.containers`. The other option would be to fail loudly.

Reply:
I would vote for later, and state that in the spec.

Reply:
I'm not convinced erroring is the best behavior in all cases. Consider the RuntimeComponent resource where there's and unnamed implicit container at the root of the spec and sets of named init/sidecar container, if we fail hard no filtering of the init and sidecar containers would be possible.

Reply:
I added a suggestion that if either field is missing, that the container should be bound.